### DISCUSSION TITLE
Node fetch abort reason missing

### DISCUSSION THREAD

**Discussion by RobinClowers** [mswjs/msw#1828] [Q&A] [answered]
I ran into an issue during my migration to 2.0, it seems that node fetch doesn't actually throw an exception object with a the abort reason when a fetch is aborted. This is important to my app, I have tests around this information. I realize this isn't an issue in MSW per se, which is why I'm not opening an issue. This used to work, so it does feel like a regression compared to 1.x. Does anyone have any suggestions? I tried switching to node-fetch, but has the same issue :(

https://github.com/nodejs/undici/pull/2289
https://github.com/node-fetch/node-fetch/pull/1775

**Comment by mattcosta7** [comment]
> I ran into an issue during my migration to 2.0, it seems that node fetch doesn't actually throw an exception object with a the abort reason when a fetch is aborted. This is important to my app, I have tests around this information. I realize this isn't an issue in MSW per se, which is why I'm not opening an issue. This used to work, so it does feel like a regression compared to 1.x. Does anyone have any suggestions? I tried switching to node-fetch, but has the same issue :(
> 
> [nodejs/undici#2289](https://github.com/nodejs/undici/pull/2289) [node-fetch/node-fetch#1775](https://github.com/node-fetch/node-fetch/pull/1775)

Do you have a small repro case, or a test that fails in v2 but passed in v1 that I can take a look at? I'm not sure offhand what the best path is here, if the underlying fetch libraries aren't doing what's expected, but more curious about where that was working beforehand - and what the difference is

**Reply by kettanaito** [reply]
I suspect the difference is mainly in using global fetch in Node vs other clients in the past? Undici indeed doesn't support a custom abort reason, so MSW 2.0 doesn't support it either. 

@RobinClowers, is it by any chance related to mocking _network errors_? As Matt mentioned, a reproduction repository is the way forward.

**Reply by RobinClowers** [reply]
I'll try to put together a repro today.

My tests are JSDOM jest tests, so they always used whatever version of fetch that is (I assume JSDOM provides this, though I'm not totally sure).

The case where I saw this was where I simulated a client side timeout, in this test I configure my client to timeout (call signal.abort) after 1ms, and the MSW handler has a promise that waits longer than that to resolve. With MSW 1.x, my catch block gets an abort exception just like in a real browser, but now the exception object is null.

**Comment by RobinClowers** [comment, marked as answer]
I was able to get around this by upgrading jest, which uses a newer JSDOM. Thanks for the responses!
